src/devices/vdml_imu.c

```diff
@@ -97,15 +105,42 @@ euler_s_t imu_get_euler(uint8_t port) {
 }
 
 double imu_get_pitch(uint8_t port) {
-	return imu_get_euler(port).pitch;
+	double rtn = PROS_ERR_F;
+	if (!claim_port_try(port - 1, E_DEVICE_IMU)) {
+		return rtn;
+	}
+	euler_s_t euler_values;
+	v5_smart_device_s_t* device;
+	device = registry_get_device(port - 1);
+	vexDeviceImuAttitudeGet(device->device_info, (V5_DeviceImuAttitude*)&euler_values);
+	rtn = euler_values.pitch + ((imu_data_s_t*)registry_get_device(port - 1)->pad)->pitch_offset;
+	return_port(port - 1, rtn);
 }
 
 double imu_get_roll(uint8_t port) {
-	return imu_get_euler(port).roll;
+	double rtn = PROS_ERR_F;
+	if (!claim_port_try(port - 1, E_DEVICE_IMU)) {
+		return rtn;
+	}
+	euler_s_t euler_values;
+	v5_smart_device_s_t* device;
+	device = registry_get_device(port - 1);
```

Comment:
Same here

Reply:
Got it, will fix for previous code as well. Should I also take a look around and fix the other sensors in a seperate PR for this issue?